core/src/createTextMaskInputElement.js

```diff
@@ -25,160 +73,159 @@ export default function createTextMaskInputElement({
     providedMask = providedMask.mask
   }
 
-  // Anything that we will need to keep between `update` calls, we will store in this `state` object.
-  const state = {previousConformedValue: emptyString}
-
   // The `placeholder` is an essential piece of how Text Mask works. For a mask like `(111)`, the placeholder would be
   // `(___)` if the `placeholderChar` is set to `_`.
   let placeholder
 
-  // We don't know what the mask would be yet. If it is an array, we take it as is, but if it's a function, we will
-  // have to call that function to get the mask array.
-  let mask
-
   // If the provided mask is an array, we can call `convertMaskToPlaceholder` here once and we'll always have the
   // correct `placeholder`.
   if (providedMask instanceof Array) {
     placeholder = convertMaskToPlaceholder(providedMask, placeholderChar)
   }
 
   return {
-    state,
-
-    // `update` is called by framework components whenever they want to update the `value` of the input element.
-    // The caller can send a `rawValue` to be conformed and set on the input element. However, the default use-case
-    // is for this to be read from the `inputElement` directly.
-    update(rawValue = inputElement.value) {
-      // If `rawValue` equals `state.previousConformedValue`, we don't need to change anything. So, we return.
-      // This check is here to handle controlled framework components that repeat the `update` call on every render.
-      if (rawValue === state.previousConformedValue) { return }
+    mask: providedMask,
+    pipe,
+    placeholder
+  }
+}
 
-      // We check the provided `rawValue` before moving further.
-      // If it's something we can't work with `getSafeRawValue` will throw.
-      const safeRawValue = getSafeRawValue(rawValue)
+export function getConformedInputState({
+  rawValue,
+  previousConformedValue,
+  currentCaretPosition,
+  mask: providedMask,
+  guide,
+  pipe,
+  placeholderChar = defaultPlaceholderChar,
+  onAccept,
+  onReject,
+  placeholder,
+  keepCharPositions = false,
+}) {
+  if (rawValue === previousConformedValue) {
```

Comment:
The previous code returned nothing. However we just keep the same signature from the `getConformedInputState` so that you don't have to do checks for undefined or not.